src/cohortMiddleware.js

```diff
@@ -0,0 +1,16 @@
+import { cohortMiddlewarePath } from './localconf';
+
+export const getAllSources = async () => {
+    const sources = await fetch(`${cohortMiddlewarePath}sources`)
+    return sources.json();
+}
+
+export const getCohortJsonByName = async (sourceName) => {
+    const cohortJson = await fetch(`${cohortMiddlewarePath}cohort-data/${sourceName}`);
+    return cohortJson.json();
+}
+
+export const getCohortCsvByName = async (sourceName) => {
+    const cohortCsv = await fetch(`${cohortMiddlewarePath}cohort-data/${sourceName}?format=csv`)
```

Comment:
you should do proper error handling for all these `fetch()` calls, can't just assume they will get a 200 everytime

maybe a good idea to check how `fetch().then().catch()` works

Reply:
added then() and catch() to middleware calls. if then() doesn't return valid response data then empty array is returned